plugins/outputs/sql/sql.go

```diff
@@ -248,15 +251,36 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 		}
 
 		sql := p.generateInsert(tablename, columns)
-		_, err := p.db.Exec(sql, values...)
 
-		if err != nil {
-			// check if insert error was caused by column mismatch
-			p.Log.Errorf("Error during insert: %v, %v", err, sql)
-			return err
+		switch p.Driver {
+		case "clickhouse":
+			// ClickHouse needs to batch inserts with prepared statements
+			tx, err := p.db.Begin()
+			if err != nil {
+				return fmt.Errorf(": %v", err)
+			}
+			stmt, err := tx.Prepare(sql)
+			if err != nil {
+				return fmt.Errorf("Error during prepare: %v", err)
+			}
+			defer stmt.Close() //nolint
+
+			_, err = stmt.Exec(values...)
+			if err != nil {
+				return fmt.Errorf("Error during execution: %v", err)
+			}
+			err = tx.Commit()
+			if err != nil {
+				return fmt.Errorf("Error during commit: %v", err)
+			}
+		default:
+			_, err = p.db.Exec(sql, values...)
+			if err != nil {
+				return fmt.Errorf("Error during execution: %v", err)
+			}
 		}
 	}
-	return nil
+	return err
```

Comment:
I think we can only end up here if there is no error so keeping `return nil` is fine.